crates/db_views/src/post_view.rs

```diff
@@ -352,40 +352,44 @@ fn queries<'a>() -> Queries<
       query = query.filter(post_aggregates::creator_id.eq(creator_id));
     }
 
-    if let Some(person_id) = person_id {
-      let is_subscribed = exists(
-        community_follower::table.filter(
-          post_aggregates::community_id
-            .eq(community_follower::community_id)
-            .and(community_follower::person_id.eq(person_id)),
-        ),
-      );
-      match options.listing_type.unwrap_or_default() {
-        ListingType::Subscribed => query = query.filter(is_subscribed),
-        ListingType::Local => {
-          query = query
-            .filter(community::local.eq(true))
-            .filter(community::hidden.eq(false).or(is_subscribed));
-        }
-        ListingType::All => query = query.filter(community::hidden.eq(false).or(is_subscribed)),
-        ListingType::ModeratorView => {
-          query = query.filter(exists(
-            community_moderator::table.filter(
-              post::community_id
-                .eq(community_moderator::community_id)
-                .and(community_moderator::person_id.eq(person_id)),
-            ),
-          ));
+    if let Some(listing_type) = options.listing_type {
```

Comment:
Logic change is here: only filter by listing_type when its provided (and it isn't from api/read_person)

Reply:
Im not sure if this is correct, looks like it could bring back https://github.com/LemmyNet/lemmy/issues/4180 which I fixed recently, or something similar. Can you add a test case to check that sorting without login works as expected?

Reply:
I think you can also do `if let (Some(listing_type), Some(person_id)) = ...` here and merge the two else branches.

Reply:
I feel like you can merge the two else branches to make it a little cleaner, but not sure how. Anyway its fine like this.